src/coreclr/src/vm/nativeimage.cpp

```diff
@@ -53,7 +53,7 @@ void NativeImage::Initialize(READYTORUN_HEADER *pHeader, LoaderAllocator *pLoade
 
     m_pReadyToRunInfo = new ReadyToRunInfo(/*pModule*/ NULL, m_pImageLayout, pHeader, /*compositeImage*/ NULL, pamTracker);
     m_pComponentAssemblies = m_pReadyToRunInfo->FindSection(ReadyToRunSectionType::ComponentAssemblies);
-    m_componentAssemblyCount = m_pComponentAssemblies->Size / sizeof(READYTORUN_COMPONENT_ASSEMBLIES_ENTRY);
+    m_componentAssemblyCount = (m_pComponentAssemblies != NULL ? m_pComponentAssemblies->Size / sizeof(READYTORUN_COMPONENT_ASSEMBLIES_ENTRY) : 1);
```

Comment:
It does not sound right to set this to `1` when there is none.

Reply:
Test for null in the place where this is used instead?

Reply:
Well, I think that to some extent this flows from our discussion when I was reviewing the R2R format doc update. I was under the impression that you were suggesting we shouldn't be creating the component assembly table when there's just one component assembly in the composite image; that's what I ended up merging in as part of the format doc and implementing in the Crossgen2 compiler. This runtime change is just a counterpart stating that, when the component assembly table doesn't exist, there's just one assembly (represented by the master R2R header) in the native image. In hindsight I'm wondering whether this special case is truly worth the effort especially as the special casing is a typical bug farm.

Reply:
I have simplified the change by removing the special-casing of composite images with just 1 component assembly. This largely simplified the relevant code and fixed some issues I was hitting in local testing.